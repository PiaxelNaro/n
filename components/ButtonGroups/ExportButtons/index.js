// @flow
import React from 'react';
import styled from 'styled-components';
import DownloadIcon from 'react-icons/lib/fa/download';
import Button from '../../Button';
import ButtonGroup from '../ButtonGroup';
import FlexGrid from '../../FlexGrid';

const IconLabelWrapper = styled(FlexGrid.Item)`
    color: #999;
    font-size: 12px;
`;

const StyledButton = styled(Button)`
    padding: 5px 6px;
    font-size: 11px;
    color: #999;
    text-decoration: none;
    &:hover,
    &:focus {
        background-image: linear-gradient(to bottom, #fff, #e2e2e2);
    }
`;

const ExportButtons = ({
    csvLink,
    xlsxLink,
}: {
    csvLink: string,
    xlsxLink: string,
}) => (
    <FlexGrid inline gutter="5px">
        <IconLabelWrapper flex={0}>
            <DownloadIcon size={12} />
        </IconLabelWrapper>
        <IconLabelWrapper flex={0}>
            Export
        </IconLabelWrapper>
        <ButtonGroup>
            <StyledButton
                target="_blank"
                rel="noreferrer noopener"
                href={csvLink}
            >
                CSV
            </StyledButton>
            <StyledButton
                target="_blank"
                rel="noreferrer noopener"
                href={xlsxLink}
            >
                XLSX
            </StyledButton>
        </ButtonGroup>
    </FlexGrid>
);

export default ExportButtons;
