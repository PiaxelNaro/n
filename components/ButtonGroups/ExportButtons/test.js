// @flow
import React from 'react';
import { shallow } from 'enzyme';
import ExportButtons from '.';

describe('<ExportButtons>', () => {
    it('renders button', () => {
        const links = {
            csv: 'testCsv',
            xlsx: 'testXlsx',
        };
        const buttons = shallow(
            <ExportButtons csvLink={links.csv} xlsxLink={links.xlsx} />
        );

        expect(buttons).toMatchSnapshot();
    });
});
