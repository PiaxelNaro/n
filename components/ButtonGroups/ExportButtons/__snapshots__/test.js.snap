// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExportButtons> renders button 1`] = `
<FlexGrid
  gutter="5px"
  inline={true}
>
  <ExportButtons__IconLabelWrapper
    flex={0}
  >
    <FaDownload
      size={12}
    />
  </ExportButtons__IconLabelWrapper>
  <ExportButtons__IconLabelWrapper
    flex={0}
  >
    Export
  </ExportButtons__IconLabelWrapper>
  <ButtonGroup>
    <ExportButtons__StyledButton
      href="testCsv"
      rel="noreferrer noopener"
      target="_blank"
    >
      CSV
    </ExportButtons__StyledButton>
    <ExportButtons__StyledButton
      href="testXlsx"
      rel="noreferrer noopener"
      target="_blank"
    >
      XLSX
    </ExportButtons__StyledButton>
  </ButtonGroup>
</FlexGrid>
`;
